src/Features/LanguageServer/Protocol/Features/Options/CompletionOptionsStorage.cs

```diff
@@ -30,31 +30,33 @@ public static CompletionOptions GetCompletionOptions(this IGlobalOptionService o
+
     // feature flags
 
-    public static readonly Option2<bool> UnnamedSymbolCompletionDisabledFeatureFlag = new("dotnet_disable_unnamed_symbol_completion", CompletionOptions.Default.UnnamedSymbolCompletionDisabled);
+    public static readonly Option2<bool> UnnamedSymbolCompletionDisabledFeatureFlag = new("dotnet_disable_unnamed_symbol_completion", CompletionOptions.Default.UnnamedSymbolCompletionDisabled, group: s_completionOptionGroup);
     public static readonly Option2<bool> ShowNewSnippetExperienceFeatureFlag = new("dotnet_show_new_snippet_experience_feature_flag", CompletionOptions.Default.ShowNewSnippetExperienceFeatureFlag);
-    public static readonly PerLanguageOption2<bool> HideAdvancedMembers = new("dotnet_hide_advanced_members_in_completion", CompletionOptions.Default.HideAdvancedMembers);
-    public static readonly PerLanguageOption2<bool> TriggerOnTyping = new("dotnet_trigger_completion_on_typing", CompletionOptions.Default.TriggerOnTyping);
-    public static readonly PerLanguageOption2<bool> TriggerOnTypingLetters = new("dotnet_trigger_completion_on_typing_letters", CompletionOptions.Default.TriggerOnTypingLetters);
-    public static readonly PerLanguageOption2<bool?> TriggerOnDeletion = new("dotnet_trigger_completion_on_deletion", CompletionOptions.Default.TriggerOnDeletion);
-    public static readonly PerLanguageOption2<EnterKeyRule> EnterKeyBehavior = new("dotnet_return_key_completion_behavior", CompletionOptions.Default.EnterKeyBehavior, serializer: EditorConfigValueSerializer.CreateSerializerForEnum<EnterKeyRule>());
+    public static readonly PerLanguageOption2<bool> HideAdvancedMembers = new("dotnet_hide_advanced_members_in_completion", CompletionOptions.Default.HideAdvancedMembers, group: s_completionOptionGroup);
+    public static readonly PerLanguageOption2<bool> TriggerOnTyping = new("dotnet_trigger_completion_on_typing", CompletionOptions.Default.TriggerOnTyping, group: s_completionOptionGroup);
+    public static readonly PerLanguageOption2<bool> TriggerOnTypingLetters = new("dotnet_trigger_completion_on_typing_letters", CompletionOptions.Default.TriggerOnTypingLetters, group: s_completionOptionGroup);
+    public static readonly PerLanguageOption2<bool?> TriggerOnDeletion = new("dotnet_trigger_completion_on_deletion", CompletionOptions.Default.TriggerOnDeletion, group: s_completionOptionGroup);
+    public static readonly PerLanguageOption2<EnterKeyRule> EnterKeyBehavior = new("dotnet_return_key_completion_behavior", CompletionOptions.Default.EnterKeyBehavior, serializer: EditorConfigValueSerializer.CreateSerializerForEnum<EnterKeyRule>(), group: s_completionOptionGroup);
     public static readonly PerLanguageOption2<SnippetsRule> SnippetsBehavior = new("dotnet_snippets_behavior", CompletionOptions.Default.SnippetsBehavior, serializer: EditorConfigValueSerializer.CreateSerializerForEnum<SnippetsRule>());
```

Comment:
>SnippetsBehavior [](http://example.com/codeflow?start=60&length=16)

Shouldn't all options in this type have the group set?

Reply:
I think they are here because the snippet internally is using the completion engine. But from the user's perspective they should belong to SnippetOptionsStorage.

Reply:
It seems though that this is specifically tweaking how snippets behave in completion lists. 
Although you can reasonably categorize it many different ways, I think it makes sense to have the option here, part of completion options and completion options group.

See SnippetsRule:

```
public enum SnippetsRule
    {
        /// <summary>
        /// Snippet triggering follows the default rules of the language.
        /// </summary>
        Default = 0,

        /// <summary>
        /// Snippets are never included in the completion list
        /// </summary>
        NeverInclude = 1,

        /// <summary>
        /// Snippets are always included in the completion list.
        /// </summary>
        AlwaysInclude = 2,

        /// <summary>
        /// Snippets are included if the user types: id?&lt;tab&gt;
        /// </summary>
        IncludeAfterTypingIdentifierQuestionTab = 3,
    }
```